sdk/index.ts

```diff
@@ -217,18 +207,22 @@ export class Flagsmith {
             return cachedItem;
         }
         traits = traits || {};
-        if (this.enableLocalEvaluation) {
-            return new Promise((resolve, reject) =>
-                this.environmentPromise!.then(() => {
-                    resolve(this.getIdentityFlagsFromDocument(identifier, traits || {}));
-                }).catch(e => reject(e))
-            );
-        }
-        if (this.offlineMode) {
-            return this.getIdentityFlagsFromDocument(identifier, traits || {});
+        try {
+            if (this.enableLocalEvaluation || this.offlineMode) {
+                const environment = await this.getEnvironment();
+                return this.getIdentityFlagsFromDocument(environment, identifier, traits || {});
+            }
+            return await this.getIdentityFlagsFromApi(identifier, traits, transient);
+        } catch (error) {
+            if (!this.defaultFlagHandler) {
+                throw new Error('getIdentityFlags failed and no default flag handler was provided')
```

Comment:
As a user, I'd prefer the original error thrown, or referenced. We can use the ES2022 way:

```suggestion
                throw new Error('getIdentityFlags failed and no default flag handler was provided', { cause: error })
```